generate.js

```diff
@@ -1,76 +1,55 @@
 'use strict';
+var d3 = require('d3-queue');
 var fs = require('fs');
-var path = require('path');
-var zlib = require('zlib');
-var tar = require('tar');
 var got = require('got');
-var cv2json = require('convert-json');
-var globby = require('globby');
-var del = require('del');
 
 // config
-var url = 'http://git.spdx.org/?p=license-list.git;a=snapshot;h=HEAD;sf=tgz';
-var folder = 'licenses';
-var ignored = [
-	'README.txt',
-	'Updating the SPDX Licenses.txt'
-];
-var colValue = 'v';
-var licensesJson = {};
+const URL = 'https://spdx.org/licenses/licenses.json';
+const MAX_CONCURRENT_CONNECTIONS = 10;
 
-// process the excel sheet and generate json files from it
-function processXls() {
-	cv2json.xls(globby.sync('licenses/spdx_licenselist_*.xls')[0], function (err, result) {
-		if (err) {
-			throw err;
-		}
+var licensesJson = {};
+var counter = 0;
+
+function downloadLicense(license, callback) {
+	got(license.detailsUrl, { json: true })
+		.then(response => {
+			var details = response.body;
+			licensesJson[license.licenseId].license = details.licenseText.replace(/\r\n/g, '\n').trim();
+			console.log(`Downloaded ${++counter} of ${Object.keys(licensesJson).length} license(s).`)
+			callback(null);
+		})
+		.catch(error => {
+			callback(`Error getting URL ${license.detailsUrl}. Response is:
+			${error.response.body}`);
+		});
+}
 
-		var licenses = result.Sheets.licenses;
-		for (var key in licenses) {
-			var row = key.match(/^A(\d+)$/);
-			if (row && row[1] > 1) {
-				var name = licenses[row[0]][colValue].trim();
-				var identifier = licenses['B' + row[1]][colValue].trim();
-				var url = licenses['C' + row[1]] ? licenses['C' + row[1]][colValue] : null;
-				var approved = licenses['E' + row[1]] ? licenses['E' + row[1]][colValue] === 'YES' : false;
+got(URL, { json: true })
+	.then(response => {
+		var spdx = response.body;
 
-				licensesJson[identifier] = {
-					name: name,
-					url: url,
-					osiApproved: approved
-				};
-			}
+		for (var license of spdx.licenses) {
+			licensesJson[license.licenseId] = {
+				name: license.name,
+				url: license.seeAlso[0], // Only get the first of possibly multiple URLs.
+				osiApproved: license.isOsiApproved
+			};
 		}
-
 		fs.writeFileSync('spdx.json', JSON.stringify(licensesJson, null, '\t'));
 
-		Object.keys(licensesJson).forEach(function (item) {
-			licensesJson[item].license = fs.readFileSync(path.join(folder, item + '.txt')).toString().trim();
-		});
-
-		fs.writeFileSync('spdx-full.json', JSON.stringify(licensesJson, null, '\t'));
-
-		// clean the `licenses` folder
-		fs.readdir(folder, function (err, files) {
-			if (err) {
-				throw err;
+		var q = d3.queue(MAX_CONCURRENT_CONNECTIONS);
```

Comment:
The limit is required by the *server* side. With unlimited concurrent connections I was seeing occasional error responses from the server.